File: sorting/src/main/java/com/sorting/selectionsort/SelectionSort.py

def issortingrequired(elementfromlowerindex,elementfromheigherindex):
    return elementfromlowerindex > elementfromheigherindex;

def swappingtheelements(elements, fromindex,toindex):
    temp = elements[fromindex]
    elements[fromindex] = elements[toindex]
    elements[toindex] = temp


def printingtheelements(elements):
    for element in range(0,len(elements)):
        print(elements[element])


def selectionsort(elements):
    position = 0;
    for i in range(0,len(elements)):
        position = i
        for j in range(position+1,len(elements)):
            if(issortingrequired(elements[position],elements[j])):
                position = j;
        swappingtheelements(elements,i,position) 
    return elements;    

if __name__ == "__main__":
    elements = [ 5, 1, 4, 2, 8 ]
    listofelements = selectionsort(elements) 
    printingtheelements(listofelements)             

File: sorting/src/main/java/com/sorting/selectionsort/SelectionSort.js
function isSortingRequired(elementFromLowerIndex, elementFromHeigherIndex) {
    return elementFromLowerIndex > elementFromHeigherIndex
}

function swappingElements(elements, fromIndex, toIndex) {
    var temp = elements[fromIndex]
    elements[fromIndex] = elements[toIndex]
    elements[toIndex] = temp
}

function selectionSort(elements) {
    for (var i = 0; i < elements.length; i++) {
        var position = i;
        for (var j = position + 1; j < elements.length; j++) {
            if (isSortingRequired(elements[position], elements[j])) {
                position = j;
            }
        }
        swappingElements(elements, i, position)
    }
    return elements;
}

function printOutElements(elements) {
    elements.forEach(element => {
        console.log(element)
    });
}

var elementsNeedToSort = [5, 1, 4, 2, 8]
var elementsSort = selectionSort(elementsNeedToSort)
printOutElements(elementsSort)

